source/shared/core_stmt.cpp

```diff
@@ -2218,145 +2223,195 @@ void get_field_as_string( _Inout_ sqlsrv_stmt* stmt, _In_ SQLUSMALLINT field_ind
+                            }
+                            else {
+                                field_len_temp -= initial_field_len;
+
+                                // Get the rest of the data.
+                                r = stmt->current_results->get_data( field_index + 1, c_type, field_value_temp + initial_field_len,
+                                    field_len_temp + extra, &dummy_field_len,
+                                    false /*handle_warning*/ TSRMLS_CC );
+                                // the last packet will contain the actual amount retrieved, not SQL_NO_TOTAL
+                                // so we calculate the actual length of the string with that.
+                                if ( dummy_field_len != SQL_NO_TOTAL )
+                                    field_len_temp += dummy_field_len;
+                                else
+                                    field_len_temp += initial_field_len;
+                            }
+
+                            if( r == SQL_SUCCESS_WITH_INFO ) {
+                                core::SQLGetDiagField( stmt, 1, SQL_DIAG_SQLSTATE, state, SQL_SQLSTATE_BUFSIZE, &len
+                                                       TSRMLS_CC );
+                            }
+
+                        } while( r == SQL_SUCCESS_WITH_INFO && is_truncated_warning( state ));
+                    }
+                    else {
+
+                        // We got the field_len_temp from SQLGetData call.
+                        field_value_temp = static_cast<char*>( sqlsrv_realloc( field_value_temp, field_len_temp + extra + 1 ));
+
+                        if ( stmt->conn->ce_option.enabled == true && stmt->current_results->odbc->cursor_type != SQLSRV_CURSOR_BUFFERED ) {
+                            // if the cursor is forward only, we have no choice but to close the cursor and open it again
+                            // but if not forward only, we can simply fetch next then fetch prior to reset the cursor
+                            if ( stmt->current_results->odbc->cursor_type == SQL_CURSOR_FORWARD_ONLY ) {
+                                // reopen the cursor
+                                core::SQLCloseCursor( stmt->current_results->odbc );
+                                core::SQLExecute( stmt );
+                                // FETCH_NEXT until the cursor reaches the row that it was at
+                                for ( int i = 0; i <= stmt->fwd_row_index; i++ ) {
+                                    core::SQLFetchScroll( stmt->current_results->odbc, SQL_FETCH_NEXT, 0 );
+                                }
+                            }
+                            else {
+                                core::SQLFetchScroll( stmt->current_results->odbc, SQL_FETCH_NEXT, 0 );
+                                core::SQLFetchScroll( stmt->current_results->odbc, SQL_FETCH_PRIOR, 0 );
+                            }
```

Comment:
Please investigate whether it's possible to refactor the above specific code for AE's max fields